### PR TITLE
chore: improve error reporting for linting failures

### PR DESCRIPTION
**Pull request by evcodes**
#### Description of changes
Made [this PR](https://github.com/aws-amplify/amplify-cli/pull/12080) to allow contributors to sort their package.json. Added a script that would check if package.json files were modified in the contributors branch, and if so run the script to fix any lint issues related to package.json files. For now it's just sorting but in the future it could have more functionality. This would ensure that a customer never gets to the sad path if they ever modify package.json files.

Happy Path:
Screenshot 2023-02-23 at 11 32 20 AM

Sad Path:
Screenshot 2023-02-23 at 11 39 38 AM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [n/a] Relevant documentation is changed or added (and PR referenced)
- [n/a] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [n/a] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sdstolworthy — DISMISSED** [review]

**Comment by evcodes** [comment]
fyi the last commit should fail. Just testing that I get the right error message in the right places ;)

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "link-win": "node ./scripts/link-bin.js packages/amplify-cli/bin/amplify amplify-dev",
     "lint-check": "yarn eslint . --ext .js,.jsx,.ts,.tsx --max-warnings=1085",
     "lint-fix": "git diff --name-only --cached --diff-filter d | grep -E '\\.(js|jsx|ts|tsx)$' | xargs eslint --fix --quiet",
-    "lint-check-package-json": "yarn eslint --no-eslintrc --config .eslint.package.json.js '**/package.json'",
+    "lint-check-package-json": "yarn eslint --no-eslintrc --config .eslint.package.json.js '**/package.json' || echo 'The dependencies in `package.json` are not sorted correctly, please run yarn sort-packages to fix this issue.'",
```
Comment (Amplifiyer) [inline package.json:30]: Is sorting the only reason this lint rule can fail?

Comment (evcodes) [reply package.json:30]: hmmmm, this is a good point. Let me take a look at the lint rules

**Comment by Amplifiyer** [comment]
> fyi the last commit should fail. Just testing that I get the right error message in the right places ;)

Did the build succeed even with expected failure? https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/16493/workflows/9341fea9-b4f7-4206-bbfa-4f2c54cc3f44/jobs/802629

**Comment by sobolk** [comment]
> > fyi the last commit should fail. Just testing that I get the right error message in the right places ;)
> 
> Did the build succeed even with expected failure? https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/16493/workflows/9341fea9-b4f7-4206-bbfa-4f2c54cc3f44/jobs/802629

Yep I think that just adding `||` operator is not right, we still need to emit non-zero exit code somehow.

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "rm-dev-link": "rimraf -f \"$(yarn global bin)/amplify-dev\"",
     "setup-dev-win": "yarn dev-build && yarn link-win && yarn link-aa-win",
     "setup-dev": "yarn dev-build && yarn rm-dev-link && yarn link-dev && yarn rm-aa-dev-link && yarn link-aa-dev",
-    "sort-packages": "yarn eslint --no-eslintrc --config .eslint.package.json.js **/package.json --fix || echo 'The dependencies in `package.json` are not sorted correctly, please run yarn sort-packages to fix this issue.'",
+    "sort-packages": "yarn eslint --no-eslintrc --config .eslint.package.json.js **/package.json --fix",
```
[the remark this answers is not in the document]
Comment (evcodes) [reply package.json]: gotcha! will implement

**Comment by evcodes** [comment]
the || made it have a zero exit code so that succeeded when it should have failed.

File: package.json [hunk 3]
```diff
@@ -77,7 +77,7 @@
     "hooks": {
       "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
       "pre-push": "yarn build-tests-changed && yarn split-e2e-tests",
-      "pre-commit": "yarn verify-commit"
+      "pre-commit": "./scripts/package-json-check.sh && yarn verify-commit"
```
Comment (evcodes) [inline package.json:80]: We are changing this from being reported in the CI to being fixed before the changes are committed.

Comment (evcodes) [reply package.json:80]: If someone runs with --no-verify, the changes can get to CI but hopefully not get into dev branch

**Review by jhockett — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**